app/services/data/get-user-role-by-username.js

```diff
@@ -0,0 +1,23 @@
+const config = require('../../../knexfile').web
+const knex = require('knex')(config)
+const Roles = require('../../constants/roles')
+
+module.exports = function (username) {
```

Comment:
Yes, the tables contain constraints on userid in the user_roles table. On the view if you specify the same user name then it will locate the user and will do an update on the role and does not add a new user.